src/runtime/Intl.cpp

```diff
@@ -1808,30 +1808,34 @@ Value Intl::getOption(ExecutionState& state, Object* options, Value property, In
     }
 }
 
-std::vector<std::string> Intl::numberingSystemsForLocale(String* locale)
+static std::vector<std::string> initAvailableNumberingSystems()
 {
-    static std::vector<std::string> availableNumberingSystems;
-
+    std::vector<std::string> availableNumberingSystems;
     UErrorCode status = U_ZERO_ERROR;
-    if (availableNumberingSystems.size() == 0) {
-        UEnumeration* numberingSystemNames = unumsys_openAvailableNames(&status);
-        ASSERT(U_SUCCESS(status));
+    UEnumeration* numberingSystemNames = unumsys_openAvailableNames(&status);
+    ASSERT(U_SUCCESS(status));
 
-        int32_t resultLength;
-        // Numbering system names are always ASCII, so use char[].
-        while (const char* result = uenum_next(numberingSystemNames, &resultLength, &status)) {
-            ASSERT(U_SUCCESS(status));
-            auto numsys = unumsys_openByName(result, &status);
-            ASSERT(U_SUCCESS(status));
-            if (!unumsys_isAlgorithmic(numsys)) {
-                availableNumberingSystems.push_back(std::string(result, resultLength));
-            }
-            unumsys_close(numsys);
+    int32_t resultLength;
+    // Numbering system names are always ASCII, so use char[].
+    while (const char* result = uenum_next(numberingSystemNames, &resultLength, &status)) {
+        ASSERT(U_SUCCESS(status));
+        auto numsys = unumsys_openByName(result, &status);
+        ASSERT(U_SUCCESS(status));
+        if (!unumsys_isAlgorithmic(numsys)) {
+            availableNumberingSystems.push_back(std::string(result, resultLength));
         }
-        uenum_close(numberingSystemNames);
+        unumsys_close(numsys);
     }
+    uenum_close(numberingSystemNames);
 
-    status = U_ZERO_ERROR;
+    return availableNumberingSystems;
+}
+
+std::vector<std::string> Intl::numberingSystemsForLocale(String* locale)
+{
+    static std::vector<std::string> availableNumberingSystems = initAvailableNumberingSystems();
```

Comment:
Vector `availableNumberingSystems` can be shared by all threads, right?
If so, don't we have to care about concurrent initialization for the vector?
(Is initialization only once guaranteed in this case?)

Reply:
Yes static function variable initialization is thread-safe.
that's why I make a function for initialization the variable.
see https://stackoverflow.com/questions/8102125/is-local-static-variable-initialization-thread-safe-in-c11